glusterfs/datadog_checks/glusterfs/check.py

```diff
@@ -0,0 +1,207 @@
+        except KeyError as e:
+            self.log.debug("Could not retrieve GlusterFS version: %s", str(e))
+
+        if raw_version:
+            version_parts = self.parse_version(raw_version)
+
+            self.set_metadata('version', raw_version, scheme='parts', part_map=version_parts)
+            self.log.debug('Found GlusterFS version: %s', raw_version)
+        else:
+            self.log.warning('Could not retrieve GlusterFS version info: %s', raw_version)
+
+    def parse_version(self, version):
+        """
+        GlusterFS versions are in format <major>.<minor>
+        """
+        major, minor = version.split('.')
+
+        return {'major': str(int(major)), 'minor': str(int(minor))}
+
+    def parse_volume_summary(self, output):
+        for volume in output:
+            volume_tags = ["vol_name:{}".format(volume['name']), "vol_type:{}".format(volume['type'])]
+            volume_tags.extend(self._tags)
+            self.submit_metric(volume, 'volume', VOLUME_STATS, volume_tags)
+
+            if 'subvols' in volume:
+                self.parse_subvols_stats(volume['subvols'], volume_tags)
+
+            self.submit_service_check(self.VOLUME_SC, volume['health'], volume_tags)
+
+    def parse_subvols_stats(self, subvols, volume_tags):
+        for subvol in subvols:
+            self.submit_metric(subvol, 'subvol', VOL_SUBVOL_STATS, volume_tags)
+
+            if 'bricks' in subvol:
+                for brick in subvol['bricks']:
+                    brick_name = brick['name'].split(":")
+                    brick_server = brick_name[0]
+                    brick_export = brick_name[1]
+                    brick_type = brick['type']
+                    brick_device = brick['device']
+                    fs_name = brick['fs_name']
+                    tags = [
+                        'brick_server:{}'.format(brick_server),
+                        'brick_export:{}'.format(brick_export),
+                        'type:{}'.format(brick_type),
+                        'device:{}'.format(brick_device),
+                        'fs_name:{}'.format(fs_name),
+                    ]
+                    tags.extend(self._tags)
+                    self.submit_metric(brick, 'brick', BRICK_STATS, tags)
+
+            self.submit_service_check(self.BRICK_SC, subvol['health'], volume_tags)
+
+    def submit_metric(self, payload, prefix, metric_mapping, tags):
+        """
+        Parse a payload with a given metric_mapping and submit metric for valid values.
+        Some values contain measurements like `GiB` which should be removed and only submitted if consistent
+        """
+        for key, metric in metric_mapping.items():
+            if key in payload:
+                value = payload[key]
+
+                if key in PARSE_METRICS:
+                    value_parsed = value.split(" ")
+                    if value_parsed[1] == "GiB":
+                        value = value_parsed[0]
+                    else:
```

Comment:
Do we expect other units to be possible?

Reply:
No I've added `-u g` to the subprocess out, so it's enforcing now.